extensions/wikia/ImageReview/modules/PromoteImage/PromoteImageReviewTask.php

```diff
@@ -1,429 +1,429 @@
+		//if images uploaded but not from import script
+			//saving changes in city_visualization_images table and purging cache
+			$this->addImagesToPromoteDb($targetWikiId, $targetWikiLang, $uploadedImages);
+			$this->model->purgeWikiPromoteDataCache($targetWikiId, $targetWikiLang);
+		}
+
+		if( !empty($uploadedImages) ) {
+		//if wikis have been added by import script or regularly by Special:Promote
+			$this->model->purgeVisualizationWikisListCache($targetWikiId, $targetWikiLang);
+		}
+
+		return !$isError;
+	}
+
+	function uploadSingleImage($imageId, $destinationName, $targetWikiId, $sourceWikiId) {
+		global $IP, $wgWikiaLocalSettingsPath;
+
+		$retval = "";
+		$imageTitle = GlobalTitle::newFromId($imageId, $sourceWikiId);
+
+		$sourceImageUrl = null;
+
+		$sourceFile = \GlobalFile::newFromText($imageTitle->getText(), $sourceWikiId);
+		if ($sourceFile->exists()){
+			$sourceImageUrl = $sourceFile->getUrl();
+		} else {
+			$this->log('Apparently the image from city_id=' . $sourceWikiId . ' ' . $imageTitle->getText() . ' is unaccessible');
+			return array('status' => 1);
+		}
+
+		$city_url = WikiFactory::getVarValueByName("wgServer", $targetWikiId);
+		if( empty($city_url) ) {
+			$this->log('Apparently the server for ' . $targetWikiId . ' is not available via WikiFactory');
+			return array('status' => 1);
+		}
+
+		$destinationName = PromoImage::fromPathname($destinationName)->ensureCityIdIsSet($sourceWikiId)->getPathname();
+
+		$sCommand = "SERVER_ID={$targetWikiId} php $IP/maintenance/wikia/ImageReview/PromoteImage/upload.php";
+		$sCommand .= " --originalimageurl=" . escapeshellarg($sourceImageUrl);
+		$sCommand .= " --destimagename=" . escapeshellarg($destinationName);
+		$sCommand .= " --wikiid=" . escapeshellarg( $sourceWikiId );
+		$sCommand .= " --conf {$wgWikiaLocalSettingsPath}";
+		
+		$logdata = [
+			'command' => $sCommand,
+			'city_url' => $city_url
+		];
+
+		$output = wfShellExec($sCommand, $retval);
+		
+		$logdata['output'] = $output;
+		$logdata['retval'] = $retval;
+		
+		if( $retval ) {
+			$this->log('Upload error! (' . $city_url . '). Error code returned: ' . $retval . ' Error was: ' . $output);
+		} else {
+			$this->log('Upload successful: '.$output);
+		}
+
+		$output = json_decode($output);
+
+		return array(
+			'status' => $retval,
+			'name' => $output->name,
+			'id' => $output->id, //page_id
+		);
+	}
+
+	public function removeImages($corpWikiLang, $wikis) {
+		$app = F::app();
+		$corpWikiId = $this->model->getTargetWikiId($corpWikiLang);
+
+		foreach($wikis as $sourceWikiId => $images) {
+			$sourceWikiLang = WikiFactory::getVarValueByName('wgLanguageCode', $sourceWikiId);
+
+			if( !empty($images) ) {
+				$removedImages = array();
+				foreach($images as $imageName) {
+					if (PromoImage::fromPathname($imageName)->isValid()) {
+						$result = $this->removeSingleImage($corpWikiId, $imageName);
+
+						if( $result['status'] === 0 ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)